microsoft/testsuites/performance/storageperf.py

```diff
@@ -187,6 +194,109 @@ def perf_storage_over_nfs_synthetic_tcp_4k(self, result: TestResult) -> None:
     def perf_storage_over_nfs_synthetic_udp_4k(self, result: TestResult) -> None:
         self._perf_nfs(result, protocol="udp")
 
+    @TestCaseMetadata(
+        description="""
+        This test case uses fio to test data disk performance with 4K block size.
+        """,
+        priority=3,
+        timeout=TIME_OUT,
+    )
+    def perf_storage_generic_fio_test(self,
```

Comment:
Can you please explain in comments how these tests are different from `perf_premium_datadisks_4k`?

Reply:
This test case is not hardcoded and run the test case as per runbook parameters. Even mode for running FIO test case is flexible and can be configured with parameters. This will provide flexibility to run FIO tests easily by just passing parameters in runbook as per need.

Whereas perf_premium_datadisks_4k seems doing same thing but with hardcoded parameters. It will launch FIO test for all 4 modes as perf_disk function run for 4 modes. Please let me know your thoughts on this.

Reply:
How about we modify `perf_premium_datadisks_4k` to work with current hardcoded setting if no parameters are specified, otherwise run it based on parameters?

Reply:
My concern is primarily to share as much code as possible. Please feel free to suggest any alternatives

Reply:
`perf_premium_datadisks_4k` has Azure specific naming as well as `requirements`. That's why we can't use it directly.
@smit-gardhariya Why not re-use the underlying `perf_disk` function? If no parameters are provided we can fallback to the configuration used by `perf_premium_datadisks_4k`.

Reply:
Sure @anirudhrb we can do it. I will make that change

Reply:
Using perf_disk for all operation having leverage of passing params from runbook as well to run FIO with different configurations. perf_disk is running all FIO mode with all these configs if passed or perf_disk will be executed with default config.